### PR TITLE
[Java] Add more OpenVINO 2.0 bindings

### PR DESCRIPTION
**Pull request by dkurt**
* Remove publication from Gradle
* Add OS specific suffix for output jar:
* Add more OpenVINO API 2.0 methods

### CODE REVIEW

File: modules/java_api/build.gradle [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+
+def nativesCPP;
+def openvinoVersion = "2022.1"
+
+if (OperatingSystem.current().isMacOsX()) {
+    nativesCPP = 'macosx-x86_64'
+} else if (OperatingSystem.current().isLinux()) {
+    nativesCPP = 'linux-x86_64'
+} else if (OperatingSystem.current().isWindows()) {
+    nativesCPP = 'windows-x86_64'
```
Comment (dkurt) [inline modules/java_api/build.gradle:16]: Need arch detection

File: modules/java_api/cpp/infer_request.cpp [hunk 2]
```diff
@@ -16,6 +16,22 @@ JNIEXPORT void JNICALL Java_org_intel_openvino_InferRequest_Infer(JNIEnv *env, j
         infer_request->infer();)
 }
 
+JNIEXPORT void JNICALL Java_org_intel_openvino_InferRequest_StartAsync(JNIEnv *env, jobject obj, jlong addr)
```
Comment (likholat) [inline modules/java_api/cpp/infer_request.cpp:19]: Is it safe to add more methods without tests?

Comment (dkurt) [reply modules/java_api/cpp/infer_request.cpp:19]: Good point! Will add some tests

**Comment by ilya-lavrenov** [comment]
Let's resolve merge conflicts and make CI green

File: modules/java_api/tests/ModelTests.java [hunk 3]
```diff
@@ -1,34 +1,84 @@
-package tests;
```
Comment (dkurt) [inline modules/java_api/tests/ModelTests.java:1]: This file initially added with Windows style line endings (CRLF). Replaced with Linux style LF.

File: modules/java_api/cpp/infer_request.cpp (hunk 2 above)
Comment (dkurt) [reply modules/java_api/cpp/infer_request.cpp:19]: Tests added. Please review

**Review by likholat — APPROVED** [review]

File: modules/java_api/tests/ModelTests.java [hunk 4]
```diff
@@ -1,34 +1,85 @@
...
+
+    @Test
+    public void testOutputsName() {
+        ArrayList<Output> outputs = net.outputs();
+        assertEquals(1, outputs.size());
+        assertEquals("Output name", "fc_out", outputs.get(0).get_any_name());
+    }
+
+    @Test
+    public void testOutputName() {
```
[the remark this answers is not in the document]
Comment (dkurt) [reply modules/java_api/tests/ModelTests.java:30]: It's for different method

File: modules/java_api/build.gradle [hunk 5]
```diff
@@ -1,22 +1,25 @@
...
+} else {
+    logger.warn('Unknown operating system!')
 }
 
+project.version = "${openvinoVersion}-${nativesCPP}"
```
Comment (dkurt) [inline modules/java_api/build.gradle:21]: @likholat, will ot be simpler for your experiment with the samples if I remove a new jar naming? So you can finish the samples building using Gradle and only after that I will check OS specific naming (ot's not critical for this PR).

Comment (likholat) [reply modules/java_api/build.gradle:21]: It's not a problem for me)

**Comment by alvoron** [comment]
@likholat are we waiting for green CI in this PR? If so, I'll ask CI team to restart the pipeline.
Could you please resolve conflict?

**Pull request closed without merging**